controllers/openshift_controller.go

```diff
@@ -120,6 +122,10 @@ func (r *KataConfigOpenShiftReconciler) Reconcile(ctx context.Context, req ctrl.
 		return ctrl.Result{}, err
 	}
 
+	if r.FeatureGates.IsEnabled(ctx, "timeTravel") {
```

Comment:
Maybe it is inferred, but we need to take disabling into account too.

If a feature is enabled by default, we need a way to disable it from the configmap.
And when the reconciler is called on a "Delete" operation, I would assume it should cause the selected options to be disabled.

Also it seems the reconciler is not called on an "Update" event, so if we modify the configmap to disable something we initially enabled, it will not happen. We would need to delete/recreate it to change the config.

This seems contradictory with the documentation:
```Feature gates can be enabled or disabled by editing the `osc-feature-gates` ConfigMap
resource.```

Reply:
Oh, Right,  however, I guess if we make sure after every **if IsEnabled==true {** to enable the feature **} else {** to disable the feature} it should keep everything idempotent, no?
In any case I'd try to look for a reference as i have no idea what is the best practice 
(also cc @pmores just in case he knows something about it as he's operators Pro)

Reply:
Yes, I think the "else" part would be required. No big deal for this example code, but we really need to keep it in mind on every feature we add.
Also we need to be careful when doing the "to disable the feature" part : this code will be called for every feature that is disabled by default, even when no configmap is involved. It needs to work when there is nothing to disable.

Reply:
> If a feature is enabled by default, we need a way to disable it from the configmap.
>  And when the reconciler is called on a "Delete" operation, I would assume it should cause the selected options to be disabled.

Currently you can disable from the configmap, you just need to set the value to `false` and use the else logic  as mentioned by @snir911 

> Also it seems the reconciler is not called on an "Update" event, so if we modify the configmap to disable something we initially enabled, it will not happen. We would need to delete/recreate it to change the config.

Nice catch, actually there is a missing queue.add . I will fix when implementing the Predicates.

Reply:
And yes, I'm open to ways to improve this. If someone has a better approach, is more than welcome.

Reply:
I don't have a better approach, I think you're on the right track.
My comments are probably not for this PR really, and more for our future selves that will have to be careful when implementing it.